include/swift/Syntax/RawSyntax.h

```diff
@@ -170,46 +170,50 @@ class RawSyntax final
   /// The \c SyntaxArena in which this node was allocated.
   SyntaxArena *Arena;
 
-  union {
-    struct {
-      /// Number of bytes this node takes up spelled out in the source code.
-      /// Always 0 if the node is missing.
-      unsigned TextLength : 32;
-      /// Whether this piece of syntax was actually present in the source.
-      unsigned Presence : 1;
-      unsigned IsToken : 1;
-    } Common;
-    enum { NumRawSyntaxBits = 32 + 1 + 1 };
-
-    // For "layout" nodes.
-    struct {
-      static_assert(NumRawSyntaxBits <= 64,
-                    "Only 64 bits reserved for standard syntax bits");
-      uint64_t : bitmax(NumRawSyntaxBits, 64); // align to 32 bits
-      /// Number of children this "layout" node has.
-      unsigned NumChildren : 32;
-      /// Total number of sub nodes, i.e. number of transitive children of this
-      /// node. This does not include the node itself.
-      unsigned TotalSubNodeCount : 32;
-      /// The kind of syntax this node represents.
-      unsigned Kind : bitmax(NumSyntaxKindBits, 8);
-    } Layout;
-
-    // For "token" nodes.
-    struct {
-      static_assert(NumRawSyntaxBits <= 64,
-                    "Only 64 bits reserved for standard syntax bits");
-      uint64_t : bitmax(NumRawSyntaxBits, 64); // align to 16 bits
-      /// The kind of token this "token" node represents.
-      const char *LeadingTrivia;
-      const char *TokenText;
-      const char *TrailingTrivia;
-      unsigned LeadingTriviaLength : 32;
-      unsigned TokenLength : 32;
-      unsigned TrailingTriviaLength : 32;
-      unsigned TokenKind : 16;
-    } Token;
-  } Bits;
+  struct CommonData {
+    /// Number of bytes this node takes up spelled out in the source code.
+    /// Always 0 if the node is missing.
+    uint32_t TextLength;
+    /// Whether this piece of syntax was actually present in the source.
+    SourcePresence Presence;
+    bool IsToken;
+  };
```

Comment:
Tell me if I'm wrong. But I think these field can and should be placed directly in `RawSyntax` like:
```cpp
class RawSyntax final: ... {
  SyntaxNodeId NodeId;
  SyntaxArena *Arena;
  uint32_t TextLength;
  SourcePresence Presence;
  bool isToken;
  
  union BitsData {
    LayoutData Layout;
    TokenData Token;
  } Bits;
};

Reply:
Yes, that’s the obvious solution. No idea why I (or whoever implemented it before me) didn’t think about that.